src/test_workflow/smoke_test/smoke_test_cluster_opensearch.py

```diff
@@ -0,0 +1,102 @@
+        self.version = self.bundle_manifest.build.version
+        self.platform = self.bundle_manifest.build.platform
+        self.arch = self.bundle_manifest.build.architecture
+        self.dist = self.bundle_manifest.build.distribution
+        self.distribution = Distributions.get_distribution(self.product, self.dist, self.version, work_dir)
+
+    def cluster_version(self) -> str:
+        return self.version
+
+    def download_or_copy_bundle(self, work_dir: str) -> str:
+        extension = "tar.gz" if self.dist == "tar" else self.dist
+        artifact_name = f"{self.product}-{self.version}-{self.platform}-{self.arch}.{extension}"
+        src_path = '/'.join([self.path.rstrip("/"), "dist", f"{self.product}", f"{artifact_name}"]) \
+            if self.path.startswith("https://") else os.path.join(self.path, "dist",
+                                                                  f"{self.product}", f"{artifact_name}")
+        dest_path = os.path.join(work_dir, artifact_name)
+
+        if src_path.startswith("https://"):
+            logging.info(f"Downloading artifacts to {dest_path}")
+            response = requests.get(src_path)
+            with open(dest_path, "wb") as file:
+                file.write(response.content)
+        else:
+            logging.info(f"Trying to copy {src_path} to {dest_path}")
+            # Only copy if it's a file
+            if os.path.isfile(src_path):
+                shutil.copy2(src_path, dest_path)
+                print(f"Copied {src_path} to {dest_path}")
+        return artifact_name
+
+    # Reason we don't re-use test-suite from integ-test is that it's too specific and not generic and lightweight.
+    def __installation__(self, work_dir: str) -> None:
+        self.distribution.install(self.download_or_copy_bundle(work_dir))
+        logging.info("Cluster is installed and ready to be start.")
+
+    # Start the cluster after installed and provide endpoint.
+    def __start_cluster__(self, work_dir: str) -> None:
+        self.__installation__(work_dir)
+        self.process_handler.start(self.distribution.start_cmd, self.distribution.install_dir, self.distribution.require_sudo)
+        logging.info(f"Started OpenSearch with parent PID {self.process_handler.pid}")
+        time.sleep(30)
+        logging.info("Cluster is started.")
+
+    # Check if the cluster is ready
+    def __check_cluster_ready__(self) -> bool:
+        url = "https://localhost:9200/"
+        logging.info(f"Pinging {url}")
+        try:
+            request = requests.get(url, verify=False, auth=("admin", "myStrongPassword123!"))
+            logging.info(f"Cluster response is {request.text}")
+            return 200 <= request.status_code < 300
+        except requests.RequestException as e:
+            logging.info(f"Request is {request.text}")
+            print(f"Cluster check fails: {e}")
```

Comment:
Please avoid adding print statements. I believe logging info is sufficient here

Reply:
Updated in the latest commit.